__tests__/endpoints.test.js

```diff
@@ -289,3 +289,55 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("Status 201 - creates a new comment in the database and returns it in the response", () => {
+    const newComment = {
+      username: "lurker",
+      body: "The first rule of fight club is.....post videos of the fights all over social media",
+    };
+
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(comment).toEqual({
+          author: "lurker",
+          body: "The first rule of fight club is.....post videos of the fights all over social media",
+          article_id: 2,
+          votes: 0,
+          comment_id: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+
+  test("Status 404 - should return an error message if the article doesn't exist", () => {
+    const newComment = {
+      username: "lurker",
+      body: "The first rule of fight club is.....post videos of the fights all over social media",
+    };
+    return request(app)
+      .post("/api/articles/999/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Article not found");
+      });
+  });
+
+  test("Status 400 - should return an error message if the wrong data type is passed as a param", () => {
+    const newComment = {
+      username: "lurker",
+      body: "The first rule of fight club is.....post videos of the fights all over social media",
+    };
+    return request(app)
+      .post("/api/articles/beans/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+});

```

Comment:
These tests are good but here's a couple of other things to think about.

What if you post a comment with a user that doesn't exist in the DB?

What if we try to post without sending the correct data?

Reply:
sorted :)